ui/components/institutional/confirm-remove-jwt-modal/confirm-remove-jwt-modal.js

```diff
@@ -0,0 +1,121 @@
+import React, { memo, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import Modal from '../../app/modal';
+import CustodyAccountList from '../../../pages/institutional/connect-custody/account-list';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { removeAccount } from '../../../store/actions';
+import withModalProps from '../../../helpers/higher-order-components/with-modal-props';
+import { Text } from '../../component-library';
+import Box from '../../ui/box';
+import {
+  BorderRadius,
+  DISPLAY,
+  TEXT_ALIGN,
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+
+const ConfirmRemoveJWT = ({
+  custodyAccountDetails,
+  accounts,
+  token,
+  hideModal,
+}) => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const [showMore, setShowMore] = useState(false);
+  const [tokenAccounts, setTokenAccounts] = useState([]);
+
+  useEffect(() => {
+    const filteredAccounts = custodyAccountDetails.filter((item) => {
+      const addressLower = item.address.toLowerCase();
+      return accounts.find((acc) => acc.address.toLowerCase() === addressLower);
+    });
+
+    const tokens = filteredAccounts
+      .map(({ address, name, labels, authDetails }) => {
+        const addressLower = address.toLowerCase();
+        const account = accounts.find(
+          ({ address: adressAcc }) => adressAcc.toLowerCase() === addressLower,
+        );
+        const balance = account && account.balance;
+        const getToken =
+          authDetails?.token ?? authDetails?.jwt ?? authDetails?.refreshToken;
+        return { address, name, labels, balance, token: getToken };
+      })
+      .filter(
```

Comment:
We can have filter first and then map so we only modify the valid elements. WdYt?

Reply:
Unfortunately, no, we need to filter after mapping. First, we want to get all the tokens from the below example and then filter by the one we passed via props. ie:

```
const getToken =
          authDetails?.token ?? authDetails?.jwt ?? authDetails?.refreshToken;
        return { address, name, labels, balance, **token**: getToken };
```

Reply:
Thanks for explaining. It’s clear 🙏🏻

Reply:
@DDDDDanica I could add first the filter and it worked with some changes. Let me know :)